packages/dd-trace/src/remote-configuration/integrity.js

```diff
@@ -0,0 +1,66 @@
+'use strict'
+
+const crypto = require('crypto')
+const { canonicalize } = require('json-canonicalize')
+
+function findSig (signatures, rcTargetsKeyId) {
+  for (const entry of signatures) {
+    if (entry.keyid === rcTargetsKeyId) {
+      if (!entry.sig) break
+      return entry.sig
+    }
+  }
+  throw new Error(`missing signature for key ${rcTargetsKeyId}`)
+}
+
+function localAtob (str) {
+  return Buffer.from(str, 'base64').toString('ascii')
+}
+
+function toDER (hex) {
+  return Buffer.concat([
+    Buffer.from('302a300506032b6570032100', 'hex'), // Static value containing OID for algorithm
+    Buffer.from(hex, 'hex')
+  ])
+}
+
+function checkIntegrity (rcTargetsKey, rcTargetsKeyId, clientGetConfigsResponse) {
+  const targets = clientGetConfigsResponse.targets
+  if (!targets) throw new Error('no field \'targets\' in clientGetConfigsResponse object')
```

Comment:
It's possible that there isn't a new targets for the tracer in the update, in that case the field won't be present in the json. I'm not positive how node handles JSON -> typed value so this might be fine, but it's worth nothing.